### DISCUSSION TITLE
Migrating from Old APIs to new APIs

### DISCUSSION THREAD

**Discussion by algo2t** [algo2t/alphatrade#58] [Ideas]
I have started migrating from old APIs to new APIs but master contract download is not available.

which is better master for getting instruments dhan or angelbroking
https://images.dhan.co/api-data/api-scrip-master.csv
https://margincalculator.angelbroking.com/OpenAPI_File/files/OpenAPIScripMaster.json

Problem with dhan is it is csv file while angel is json file.

**Comment by ronaldpaul** [comment]
@algo2t Can't we use Alice Blue API for getting contracts. That one appears to be a well-maintained and stable one with proper documentation? You effort in rebuilding this library is greatly appreciated. Thanks a lot for your effort. If any help needed, kindly let me know. I want this library to be a well-maintained one that follows the official API endpoints provided by SAS Online at https://drive.google.com/file/d/15HMf1ieewe2FgHxZJ-YIGeDtyNVEgCQ2/view so that this library also remains a stable one like Alice Blue. Thanks.

**Comment by ronaldpaul** [comment]
Hi @algo2t , is it not possible to utilize AliceBlue master contract shown here https://v2api.aliceblueonline.com/contractMaster?

**Reply by rahulmr** [reply]
This looks good to me. Let me check if possible to implement.

**Reply by ronaldpaul** [reply]
Bro, in the pyoauthbridge library provided by SAS, they use https://alpha.sasonline.in/api/v2/contracts.json?exchanges={exchange} for getting master contract. So I think no need to use Alice Blue. I tested and it works.

**Comment by nevatia** [comment]
> I have started migrating from old APIs to new APIs but master contract download is not available.
> 
> which is better master for getting instruments dhan or angelbroking https://images.dhan.co/api-data/api-scrip-master.csv https://margincalculator.angelbroking.com/OpenAPI_File/files/OpenAPIScripMaster.json
> 
> Problem with dhan is it is csv file while angel is json file.

master contract file download is available for sasonline. I found You ommitted something or  was it intentional to get mastercontracts from outside??

btw update line 1041
    'master_contract', Requests.GET, {'exchange': exchange}, None)
    
  Bhai basket order ka kuch mila ? ;)

**Reply by nevatia** [reply]
@algo2t rewriting code to v1 from v2, is'nt it going from new to old? What made you think doing so? am i missing something? Not placed orders in recent weeks though, else is working fine for me.

**Reply by ronaldpaul** [reply]
In which URL the master contract is available? In the old URL that @algo2t  was using in the library, the contract was not available. That is why he was asking from where we shall download it.

**Reply by yugavarmank** [reply]
Hi bro @algo2t  and guys i am new to algo trading. Pls guide me in this. Please discuss with my whatsapp number nine double eight four seven double zero three five seven. And i am learning python now. I am in sharemarket trading for last 8 years. so help me in algo trading

**Reply by rahulmr** [reply]
master_contract endpoint is giving empty data, that why this discussion. And I messed up with the repo, by mistake committed to main repo instead of dev-v1

**Reply by nevatia** [reply]
@rahulmr Dont know how its giving you empty data? Have you corrected line as told above? Try it.

**Reply by algo2t** [reply]
Now corrected, will commit in sometime

**Comment by ronaldpaul** [comment]
> master_contract endpoint is giving empty data, that why this discussion. And I messed up with the repo, by mistake committed to main repo instead of dev-v1

Even I was wondering why the commit was done in main branch instead of dev-v1. You can revert back and resolve right @rahulmr ? I have been working on https://github.com/TradelabTechnologiesIndia/pyoauthbridge since I am heavily trading using your AlphaTrade library and for the past one week, I could not do anything. You can have a look at that too and see whether it is possible to utilize that for AlphaTrade. Thanks a lot for your effort and support.

**Reply by rahulmr** [reply]
Working on it already.

**Reply by algo2t** [reply]
Are you working on websocket or only order placement? I was thinking of creating a simpler API for only order placement.

**Reply by ronaldpaul** [reply]
I already started working by changing my logic to utilize pyoauthbridge. I am able to connect and stream websocket tick for Nifty etc. I will pull your changes and see whether it works properly without anything major to be changed in my old code.

**Reply by algo2t** [reply]
It seems like pyoauthbridge is not in good shape, I would suggest you to check my current implementation and see if it works as per your need. meanwhile I will try to work on pyoauthbridge. Have you received your api_secret from SASOnline support team?

**Reply by ronaldpaul** [reply]
Yes. But made multiple changes to it based out of some code from AlphaTrade and got some functionalities like token, connection, instrument fetch, balance working. Just refactoring few more function parameters.

**Reply by algo2t** [reply]
can you share the code in some repository for connection to websocket, subscribe and unsubscribe?

**Comment by algo2t** [comment]
Made the changes but not able to test data streaming and MCX since I do not have MCX enabled.
Please test at your end and let me know.
